test/data/test_xr.py

```diff
@@ -22,15 +22,56 @@
 import numpy as np
 import pytest
 import xarray as xr
+from zarr.storage import MemoryStore
 
 from earth2studio.data import (
     DataArrayDirectory,
     DataArrayFile,
     DataArrayPathList,
     DataSetFile,
+    ModelOutputDatasetSource,
 )
 
 
+def build_model_output_dataset(
```

Comment:
Any chance this could use an IO backend?
Like so it minics the output of an inference pipeline, I feel like this would really helm make the tests check the source with how it would be used.